coastguard/coastguard.md

```diff
@@ -0,0 +1,138 @@
+# Submariner Multi-cluster network policy Support
+
+## Summary
+The purpose of this project is to introduce multi-cluster network policy to a submariner cluster.
+
+Kubernetes Network Policy allows cluster administrators to configure granular isolation rules using Network Policy objects. A network policy is a specification of how groups of pods (using labels) are allowed to communicate with each other and other network endpoints.
+
+The purpose of this project is to provide a mechanism for Network Policies across a cluster-set such that policy enforcement from a single cluster will seamlessly expand and work across clusters.
+
+Currently, in a submariner environment which includes multiple k8s clusters, there is no way to define network policy between the two clusters. An administrator must manually collect ranges of ip addresses if he wants to limit traffic. Given the kubernetes pod network model, which is a flat network, this is not granular enough. 
+
+The proposed service, named Coastguard, will allow cluster administrators to define multi-cluster network policies, and coastguard will translate the multi-cluster policies into relevant single-cluster network policies, and will keep those policies up to date after new deployments in each cluster. 
+
+## Motivation
+With the advent of [multi cluster services api](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api) the need for restricting traffic across a cluster-set has become more apparent.
+
+When a service is imported into a cluster, the imported service and it's underlying pods are accessible from all pods in the importing cluster. 
+
+Limiting traffic to the service and it's underlying pods by other pods can't be done by semantic rules (namespaces, labels), which greatly limits the options for a cluster administrator wanting to limit and secure his cross-service traffic.
+## Requirements
+1. Connected Clusters without any network policies should keep the default behavior.
+2. Allow limiting traffic to/and from a service based on namespace.
+3. Allow limiting traffic to/and from the underlying pods of a service based on labels and/or namespace.
```

Comment:
Is this just for pods underlying a service? What about just pods?

IIUC K8s policies work on pods even when no service is in front of them.